### PR TITLE
pythonPackpages.tensorflow: 0.8.0 -> 0.9.0

### PR DESCRIPTION
**Pull request by houqp**
###### Motivation for this change

update tensorflow to 0.9.0 and fix the build
###### Things done
- [x] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [ ] OS X
  - [x] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by houqp** [comment]
Hello, could someone explain to me how to test with build-use-chroot? I am not able to find the `nix.conf` file in my system. I am using nix in RHEL.

**Comment by FRidh** [comment]
@houqp it is typically `/etc/nix/nix.conf`

**Comment by FRidh** [comment]
The protobuf changes need to be in a separate commit.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -28404,15 +28404,15 @@ in modules // {
...
     format = "wheel";
 
     src = pkgs.fetchurl {
       url = "https://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-${version}-cp27-none-linux_x86_64.whl";
-      sha256 = "07lb6rknngq9bicd7z1q9caiqxlqn4fdx8q24s3rqvv9wi79szws";
+      sha256 = "15v7iyry8bmp5wcc1rr4bkp80f3887rl99zqf8pys5bad4gldbkh";
     };
 
-    propagatedBuildInputs = with self; [ numpy six protobuf3_0 pkgs.swig ];
+    propagatedBuildInputs = with self; [ numpy six ((self.protobufBuild pkgs.protobuf3_0_0b2).override { doCheck = false; }) pkgs.swig ];
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: why do you disable the tests? I suggest you add a comment about that and use a let expression.

Comment (houqp) [inline pkgs/top-level/python-packages.nix]: I copied it from the 3_0 version, which does not have comment to explain why. So honestly, I don't know the answer ;P

**Comment by houqp** [comment]
I tried manually creating /etc/nix/nix.conf and setting the build-use-chroot option. Not sure if it makes any difference though. At least the build still passes :) @FRidh could you review this again?

**Comment by houqp** [comment]
This should fix https://github.com/NixOS/nixpkgs/issues/16624

**Comment by houqp** [comment]
OSX build timed out, but I can't find any error in the build log. Is this expected?

**Comment by FRidh** [comment]
is there a need for refactoring protobuf? Can't you override it to change it to beta?

**Comment by houqp** [comment]
Could you point me to an example that overrides an existing package for another version? That was the first thing I tried, but I can't get it to work :(

**Comment by houqp** [comment]
ping @FRidh  :)

**Comment by houqp** [comment]
rebased on the latest master and added tflearn.

**Comment by zimbatm** [comment]
Would it be possible to only keep one 3.x release to keep the complexity low? I'm lacking a lot of context and don't know what 3.0.0b2 represents over the 3.0.0 stable release.

**Comment by FRidh** [comment]
@zimbatim tensorflow requires this version. @houqp , have you tried modifying `setup.py` to remove the dependency on this exact version so it would use the release instead?

**Comment by houqp** [comment]
Yes, 3.0.0b2 is a hardcoded dependency. Can we change the dependency in wheel? tensorflow is not built with non-standard google tooling so we can't change setup.py and build it from scratch.

**Comment by FRidh** [comment]
> Yes, 3.0.0b2 is a hardcoded dependency. Can we change the dependency in wheel? tensorflow is not built with non-standard google tooling so we can't change setup.py and build it from scratch.

You're right, I forgot about that. So we will really need this extra version.

**Comment by houqp** [comment]
@FRidh @zimbatm is there any example on reusing an existing package for another version? If so, we can get rid of `generic.3.nix`.

**Comment by FRidh** [comment]
@houqp I had a look at it and `overrideDerivation` is a possibility here. On second though, I think it is best here if you just copy the current expression of 3.0.nix and rename it to say 3.0.0-beta-2.nix. That way, once tensorflow is up to date again, we can simply remove it.

**Comment by houqp** [comment]
@FRidh done :)

**Comment by FRidh** [comment]
Great, thanks!

**Pull request merged**